### DISCUSSION TITLE
Template Siberia does not work with primefaces 14

### DISCUSSION THREAD

**Discussion by silviogs** [primefaces/community#2417] [PrimeFaces Templates] [answered]
Hi

I purchased the Siberia template in 2022 and it doesn't work with PrimeFaces 14.

when I bought it 
warned these terms:

    Non Commercial Usage
    Single End Product, No Multi-Use
    Lifetime Support
    Unlimited Updates

could anyone help?

thank you very much

**Comment by melloware** [comment]
@silviogs can you send me a DM on the Discord channel.  I can help you fix your Siberia template for PF14

PrimeFaces Discord: https://discord.gg/gzKFYnpmCY

**Comment by silviogs** [comment]
Hi

Dashboard.java
        List<Number> values = new ArrayList<>();
        for (int i = 0 ; i < this.graficoAnualAlegueislList.size(); i++) {
        	values.add((Number) this.graficoAnualAlegueislList.get(i)[1]);
        }  
        
        barDataSet.setData(values);  error on this line

SiberiaMenuRenderer.java

        builder.init()
                .source(clientId)
                .form(SearchExpressionFacade.resolveClientId(context, component, source.getForm())) error on this line
                .process(component, source.getProcess())
                .update(component, source.getUpdate())
                .async(source.isAsync())
                .global(source.isGlobal())
                .delay(source.getDelay())
                .timeout(source.getTimeout())
                .partialSubmit(source.isPartialSubmit(), source.isPartialSubmitSet(), source.getPartialSubmitFilter())
                .resetValues(source.isResetValues(), source.isResetValuesSet())
                .ignoreAutoUpdate(source.isIgnoreAutoUpdate())
                .onstart(source.getOnstart())
                .onerror(source.getOnerror())
                .onsuccess(source.getOnsuccess())
                .oncomplete(source.getOncomplete())
                .params(component);

thank you very much

**Reply by melloware** [reply, marked as answer]
Yep DM me on discord. It's not a simple change and I have fixed other premium templates like Ultima and Rain for others.

**Comment by tandraschko** [comment]
as melloware fixed this with you, i will close this for now

in general we decided to only maintain following templates, which were released yesterday:

          - "apollo"
          - "atlantis"
          - "avalon"
          - "diamond"
          - "freya"
          - "poseidon"
          - "ultima"
          - "verona"
